pkg/controller/capacity/capacity_controller.go

```diff
@@ -168,116 +172,122 @@ func (c *Controller) processCapacityTargetOnCluster(
+		sadPods           []shipper.PodStatus
+		reports           []shipper.ClusterCapacityReport
+	)
+
 	defer func() {
-		c.reportClusterCapacityConditionChange(ct, diff)
+		status.SadPods = sadPods
+		status.Reports = reports
+		status.AvailableReplicas = availableReplicas
+		status.AchievedPercent = c.calculatePercentageFromAmount(
+			spec.TotalReplicaCount, availableReplicas)
+
+		diff.Append(capacityutil.SetClusterCapacityCondition(status, *operationalCond))
+		diff.Append(capacityutil.SetClusterCapacityCondition(status, *readyCond))
+		c.reportConditionChange(ct, ClusterCapacityConditionChanged, diff)
 	}()
 
-	informerFactory, err := c.clusterClientStore.GetInformerFactory(spec.Name)
+	appName := ct.Labels[shipper.AppLabel]
+	release := ct.Labels[shipper.ReleaseLabel]
+	deployment, pods, err := c.getClusterObjects(spec.Name, ct.Namespace, appName, release)
 	if err != nil {
-		cond := capacityutil.NewClusterCapacityCondition(
+		operationalCond = capacityutil.NewClusterCapacityCondition(
 			shipper.ClusterConditionTypeOperational,
 			corev1.ConditionFalse,
-			ServerError,
+			InternalError,
 			err.Error())
-		diff.Append(capacityutil.SetClusterCapacityCondition(status, *cond))
 
 		return err
 	}
 
-	selector := labels.Set(ct.Labels).AsSelector()
-	targetDeployment, err := c.findTargetDeploymentForClusterSpec(informerFactory, *spec, ct.Namespace, selector, status)
-	if err != nil {
-		cond := capacityutil.NewClusterCapacityCondition(
-			shipper.ClusterConditionTypeReady,
-			corev1.ConditionFalse,
-			ServerError,
-			err.Error())
-		diff.Append(capacityutil.SetClusterCapacityCondition(status, *cond))
-
-		return err
-	}
+	operationalCond = capacityutil.NewClusterCapacityCondition(
+		shipper.ClusterConditionTypeOperational,
+		corev1.ConditionTrue,
+		"",
+		"")
 
-	var availableReplicas, achievedPercent int32
-	defer func() {
-		status.AchievedPercent = achievedPercent
-		status.AvailableReplicas = availableReplicas
-	}()
-
-	if targetDeployment == nil {
-		return nil
-	}
+	report := buildReport(deployment.Name, pods)
 
-	rCnt := int32(replicas.CalculateDesiredReplicaCount(uint(spec.TotalReplicaCount), float64(spec.Percent)))
+	// availableReplicas and reports will be used by the defer at the top
+	// of this func
+	availableReplicas = deployment.Status.AvailableReplicas
+	reports = []shipper.ClusterCapacityReport{*report}
 
-	if targetDeployment.Spec.Replicas == nil || rCnt != *targetDeployment.Spec.Replicas {
-		_, err = c.patchDeploymentWithReplicaCount(ct, targetDeployment, spec.Name, rCnt, status)
+	desiredReplicas := int32(replicas.CalculateDesiredReplicaCount(uint(spec.TotalReplicaCount), float64(spec.Percent)))
+	if deployment.Spec.Replicas == nil || desiredReplicas != *deployment.Spec.Replicas {
+		_, err = c.patchDeploymentWithReplicaCount(deployment, spec.Name, desiredReplicas)
 		if err != nil {
+			readyCond = capacityutil.NewClusterCapacityCondition(
+				shipper.ClusterConditionTypeOperational,
```

Comment:
This might be a mistake, readyCond  = ... ClusterConditionType*Operational* ... ?